extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+
+	public function getRequestData() {
+		return $this->mRequestData;
+	}
+
+	public function getOldUsername() {
+		return $this->mOldUsername;
+	}
+
+	public function getNewUsername() {
+		return $this->mNewUsername;
+	}
+
+	public function getReason() {
+		return $this->mReason;
+	}
+
+	public function getOldUid() {
+		return $this->mUserId;
+	}
+
+	public function getErrors() {
+		return $this->mErrors;
+	}
+
+	public function getWarnings() {
+		return $this->mWarnings;
+	}
+
+	public function getPhalanxBlockID() {
+		return $this->mPhalanxBlockId;
+	}
+
+	public function getUserRenameTaskId() {
+		return $this->mUserRenameTaskId;
+	}
+
+	/**
+	 * Sets destination for all the logs
+	 *
+	 * @param $destination string/enum One of RenameUserProcess::LOG_* constant
+	 * @param $task BatchTask (Optional) BatchTask to send logs to
+	 */
+	public function setLogDestination( $destination, $task = null ) {
+		$this->mLogDestinations = array();
+		$this->addLogDestination( $destination, $task );
+	}
+
+	/**
+	 * Adds another log destination
+	 *
+	 * @param $destination string/enum One of RenameUserProcess::LOG_* constant
+	 * @param $task BatchTask (Optional) BatchTask to send logs to
+	 */
+	public function addLogDestination( $destination, $task = null ) {
+		$this->mLogDestinations[] = array( $destination, $task );
+	}
+
+	/**
+	 * Saves passed error for future retrieval
+	 *
+	 * @param $msg string Error message
+	 */
+	public function addError( $msg ) {
+		$this->mErrors[] = $msg;
+	}
+
+	/**
+	 * Saves passed warning for future retrieval
+	 *
+	 * @param <type> $msg
+	 */
+	public function addWarning( $msg ) {
+		$this->mWarnings[] = $msg;
+	}
+
+	protected function getUserTableName( $database ) {
+		return "`{$database}`.`user`";
+	}
+
+	protected function renameAccount() {
+		global $wgExternalSharedDB;
+
+		$dbw = wfGetDb( DB_MASTER, array(), $wgExternalSharedDB );
+
+		$table = '`user`';
+		$this->addLog( "Changing user {$this->mOldUsername} to {$this->mNewUsername} in {$wgExternalSharedDB}" );
+
+		if ( $dbw->tableExists( $table ) ) {
+			$dbw->update( $table,
+				array( 'user_name' => $this->mNewUsername ),
+				array( 'user_id' => $this->mUserId ),
+				__METHOD__
+			);
+
+			$affectedRows = $dbw->affectedRows();
+			$this->addLog( 'Running query: ' . $dbw->lastQuery() . " resulted in {$affectedRows} row(s) being affected." );
+
+			if ( $affectedRows ) {
+				$dbw->commit();
+				$this->addLog( "Changed user {$this->mOldUsername} to {$this->mNewUsername} in {$wgExternalSharedDB}" );
+
+				User::clearUserCache( $this->mUserId );
+
+				return true;
+			} else {
+				$this->addLog( "No changes in {$wgExternalSharedDB} for user {$this->mOldUsername}" );
+			}
+		} else {
+			$this->addLog( "Table \"{$table}\" not found in {$wgExternalSharedDB}" );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if the request provided to the constructor is valid.
+	 *
+	 * @return bool True if all prerequisites are met
+	 */
+	protected function setup() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "setup" is 29 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)

[project-id]: wikia:app
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 11, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[project-id]: wikia:app
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "setup" has 191 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)

[project-id]: wikia:app